AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -357,8 +430,35 @@ - (void)_deleteNodesAtIndexPaths:(NSArray *)indexPaths withAnimationOptions:(ASD
   [self deleteNodesOfKind:ASDataControllerRowNodeKind atIndexPaths:indexPaths completion:^(NSArray *nodes, NSArray *indexPaths) {
     ASDisplayNodeAssertMainThread();
     
+    // TODO - Filter out Move indexPaths
+    NSMutableArray *newNodes = [[NSMutableArray alloc] initWithCapacity:nodes.count];
+    NSMutableArray *newIndexPaths = [[NSMutableArray alloc] initWithCapacity:indexPaths.count];
+    
+    for (int i = 0; i < indexPaths.count; i++) {
+      if (![_moveFromIndexPaths containsObject:indexPaths[i]]) {
```

Comment:
Agree with moving to `_ASHierarchyChangeSet` as the primary place handling changing data, and I want to clarify couple points before the move.

1. Should the `move` data set be consolidated on the editing queue? As it seems in `reload` case, it was split into `insert` and `delete` on the main queue.
2. Would it make sense to override some methods in `ASChangeSetDataController` to handle the `move` behaviour? (For instance, `insertNodes:` and `deleteNodesOfKind:`)
3. Instead of passing `context`, these methods can access `_changeSet` directly inside `ASChangeSetDataController`